src/server/global_state.rs

```diff
@@ -95,8 +95,15 @@ impl GlobalState {
         self.sender.send(message).unwrap()
     }
 
-    pub(crate) fn update_configuration(&mut self, _config: Config) {
-        // TODO
+    fn update_project_manifest(&mut self) {
+         ProjectManifest::discover_all(self.config.root_path());
+    }
+
+    // update the workspace data - called when (to be yet added) cargo watch discovers changes
+    pub(crate) fn update_workspace_data(&mut self) { 
+        self.update_project_manifest();
+        //get a manifest path from config and pass it to new Project Workspace
+        self.workspace = Arc::new(ProjectWorkspace::from(self.config.workspace_manifest.file.clone()));
```

Comment:
Czy nie powinniśmy czegoś robić z wynikiem funkcji discover_all? I czy potrzebujemy wydzielenia tego do funkcji update_project_manifest?

Reply:
Sure you right